Put song title text view into a horizontal scrollview so you can see the full song title on songs with longer song titles

theduffmaster opened the pull request:
There's not a lot of room for the song title in the current song view, which means the song title gets cut off most of the time and it's hard to figure out what the full name of the song is without going into the song details dialog. All of this is very annoying. Putting the textview into a horizontal scroll view is an elegant solution because it doesn't compromise any existing functionality, it doesn't change anything and it is very easy for the user to interact with.

niririke commented:
I think I found a problem. Playing a song from e.g. album or playlist does not work sometimes with this change. I think it is because when you tap song's title it tries to scroll it horizontally. This happens even if the title is not long and can't actually be scrolled. Maybe we can use separate xml for current song view because the same file (item_list.xml) is used in both current song view and in playlists and albums (do I understand this correctly?). Though pressing song's title in current song view does not open queue now too.

theduffmaster commented:
You're more then right. Actually, it seems like you can't play any song without clicking something that's not the textview. I'll have to see how I can fix this.

theduffmaster commented:
Making the text scrollable took away the ripple animations and it’s ability to field the onclicklistener, so I had to recreate and implement ripple animations and on touch from scratch. It works, although not as smoothly as I’d like. Test it out and let me know your thoughts. I’m not sure there’s an easier way to do this, but someone can try and find one.

niririke commented:
It surely works now! But there is a little (around half a second I guess) delay before ripple animation and actual song playback when I press song's title. Pressing anywhere else works fine (fast) though. It's not that big a deal I guess, but it is still perceptible.

theduffmaster commented:
@niririke I agree. I don't think we should merge this until we get it so it's seamless. I am going to try to figure out what's slowing it down.

arkon commented:
I imagine it's something to do with it trying to register what type of touch it was (i.e. Was it actually a double touch? Better wait a split second to be sure.). Is it possible to add a listener to the ScrollView so that if the scrolling motion is under a certain threshold, it just dispatches the event (tap) to the underlying view instead?

theduffmaster commented:
@arkon that's exactly what I'm trying right now.

theduffmaster commented:
I GOT IT! Should be up and working soon. Had to create a custom FrameLayout class and intercept the touch event and control whether it gets passed to the scrollview or not.

theduffmaster commented:
That was terrible. It should work smoothly now, almost as if there's no difference. I'll merge it in a couple days once a couple people have reviewed it.

theduffmaster commented:
So it seems that if you click on the 3 dots to open the menu sometimes the click also registers on the framelayout and opens the song. Gonna have to figure that out tomorrow.

arkon commented on .gitignore:
```diff
@@ -40,3 +40,7 @@ captures/
 # Mac
 .DS_Store
+
+*.gradle
```
These shouldn't be in the gitignore.

theduffmaster replied:
noted

arkon commented on build.gradle:
```diff
@@ -1,9 +1,13 @@
 buildscript {
     repositories {
         jcenter()
+        maven {
```
You should just leave this out of this PR for now.

arkon commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,6 +1,6 @@
-#Tue Apr 11 17:46:33 SGT 2017
+#Wed Jun 07 15:02:52 EDT 2017
```
You should just leave this out of this PR for now.

arkon commented on app/src/main/res/layout/item_list.xml:
```diff
@@ -20,7 +20,16 @@
         android:tintMode="src_in"
         tools:ignore="ContentDescription" />
 
+    <View
```
So is this just a dummy view for intercepting touches?

theduffmaster replied:
This is just a regular framelayout, but I've Overrided onInterceptTouch so I can delegate any touches to the right views. In this case either the HorizontalScrollView or this FrameLayout.

theduffmaster commented:
It's now 100% working with no drawbacks compared to how the app performs without the scrolling textviews. However, if you test it let me know if you find anything wrong.

arkon commented:
I think I'm expecting it to register as a regular tap (or other things like long pressing) if I just tap directly on the scrolling textview, but it doesn't. Seems like an unexpected behaviour?

theduffmaster commented:
It should just register as a regular tap, no different then if you were to touch another part of the screen.  It should only not register a tap if you start scrolling and using the scrollview. That certainly it weird, on my end it seems to be working. @arkon what activity are you in?

arkon commented:
@theduffmaster It seems to happen everywhere (e.g. album view, songs list, now playing queue).

theduffmaster commented:
@arkon so when you tap on the scrolling textview nothing happens? No animation, no click?

kabouzeid commented:
I'll have a look at this now too

theduffmaster commented:
That would be much appreciated @kabouzeid

arkon commented:
@theduffmaster Here's [a video](https://goo.gl/photos/MGmoqjiuRYafcg439).

kabouzeid commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,6 +1,6 @@
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-3.5-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.0-milestone-1-all.zip
```
Not yet in the production branch

theduffmaster replied:
You're right. My git configuration got messed up, I'll remove these

kabouzeid commented on build.gradle:
```diff
@@ -1,9 +1,13 @@
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.3.1'
+        classpath 'com.android.tools.build:gradle:3.0.0-alpha3'
```
Not yet in the production branch

theduffmaster commented:
@arkon That's not how it's supposed to act at all. That's how it acted back on the first commit. Are you sure you've synced the most recent commits?

arkon commented:
@theduffmaster Yeah, I'm on the latest commit which is weird. The only thing I changed locally was reverting the Gradle version.

kabouzeid commented:
On my end it doesn't scroll at all.

kabouzeid commented on app/src/main/java/com/kabouzeid/gramophone/views/CustomFrameLayout.java:
```diff
@@ -0,0 +1,74 @@
+        super(context, attrs, defStyleAttr);
+    }
+
+    public void getScrollView(HorizontalScrollView view){
```
Should be setScrollView()

theduffmaster replied:
Agreed.

theduffmaster commented:
I'm going to make a video just to show you what it should look like.

niririke commented:
I am experiencing the same behavior @arkon does. Noted an interesting thing: tapping the same song repeatedly very fast seems to lead to "Couldn't play this song" message and sometimes to app crash.

arkon commented:
Can confirm about the repeated tapping thing that @niririke reported, although that happens in the production version too. One thing to note is that the ripples are noticeably laggier when tapping quickly/repeatedly with these changes, probably because they're programatically triggered.

niririke commented:
@arkon Strange, I don't get the message in the production version.

theduffmaster commented:
[Here's a video of how things are looking on my end](https://gfycat.com/ImpressionableFlashyAustraliankelpie)
Obviously there's a problem, but on my end I'm not having any of these problems. Everything seems to be working perfectly. As a result, I don't even know where I would start looking for solutions. Can anybody think of a possible cause?

theduffmaster commented:
Maybe try File->Invalidate Caches and Restart and Android Studio? I know I had to do that a couple times throughout the process of building this for various reasons.

arkon commented:
@theduffmaster No luck, unfortunately.

theduffmaster commented:
@arkon @kabouzeid @niririke I added some logging. Could you send me what the logcat looks like when you touch the unresponsive scrollview?

arkon commented:
Regular tap on scroll view:
```
06-09 17:13:11.213 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_DOWN, actionButton=0, id[0]=0, x[0]=816.2442, y[0]=56.460815, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=44757147, downTime=44757147, deviceId=5, source=0x1002 }
06-09 17:13:11.214 21164-21164/com.kabouzeid.gramophone.debug D/ACTION_DOWN?: True
06-09 17:13:11.214 21164-21164/com.kabouzeid.gramophone.debug D/Event X: 816
06-09 17:13:11.214 21164-21164/com.kabouzeid.gramophone.debug D/Event Y: 696
06-09 17:13:11.214 21164-21164/com.kabouzeid.gramophone.debug D/View Left: 0
06-09 17:13:11.214 21164-21164/com.kabouzeid.gramophone.debug D/View Right: 0
06-09 17:13:11.214 21164-21164/com.kabouzeid.gramophone.debug D/View Top: 0
06-09 17:13:11.214 21164-21164/com.kabouzeid.gramophone.debug D/View Bottom: 0
06-09 17:13:11.214 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:11.251 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=816.2442, y[0]=56.460815, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=44757175, downTime=44757147, deviceId=5, source=0x1002 }
06-09 17:13:11.252 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:11.252 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:11.253 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_UP, actionButton=0, id[0]=0, x[0]=816.2442, y[0]=56.460815, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=44757188, downTime=44757147, deviceId=5, source=0x1002 }
06-09 17:13:11.253 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
```

Long press on scroll view:
```
06-09 17:13:42.287 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_DOWN, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=44788222, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.287 21164-21164/com.kabouzeid.gramophone.debug D/ACTION_DOWN?: True
06-09 17:13:42.288 21164-21164/com.kabouzeid.gramophone.debug D/Event X: 821
06-09 17:13:42.288 21164-21164/com.kabouzeid.gramophone.debug D/Event Y: 619
06-09 17:13:42.288 21164-21164/com.kabouzeid.gramophone.debug D/View Left: 0
06-09 17:13:42.288 21164-21164/com.kabouzeid.gramophone.debug D/View Right: 0
06-09 17:13:42.288 21164-21164/com.kabouzeid.gramophone.debug D/View Top: 0
06-09 17:13:42.288 21164-21164/com.kabouzeid.gramophone.debug D/View Bottom: 0
06-09 17:13:42.288 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:42.351 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=44788278, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.351 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:42.351 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:42.366 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=1, eventTime=44788299, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.366 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:42.367 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:42.382 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=1, eventTime=44788313, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.382 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:42.383 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:42.399 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=1, eventTime=44788327, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.399 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:42.399 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:42.414 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=44788348, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.414 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:42.415 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:42.430 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=1, eventTime=44788366, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.430 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:42.431 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:42.462 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=44788389, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.463 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:42.463 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:42.478 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=1, eventTime=44788410, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.478 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:42.479 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:42.494 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=1, eventTime=44788430, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.495 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:42.495 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:42.510 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=2, eventTime=44788446, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.510 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:42.510 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:42.575 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=44788501, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.575 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:42.575 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:42.591 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=1, eventTime=44788522, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.591 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:42.592 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:42.606 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=1, eventTime=44788542, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.606 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:42.606 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:42.622 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=1, eventTime=44788550, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.622 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:42.622 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:42.638 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=44788571, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.638 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:42.638 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:42.656 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=1, eventTime=44788590, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.656 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:42.656 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:42.712 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=44788640, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.712 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:42.712 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:42.732 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=1, eventTime=44788668, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.732 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:42.733 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:42.743 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=1, eventTime=44788675, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.743 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:42.744 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:42.764 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=2, eventTime=44788700, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.764 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:42.765 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:42.806 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=44788738, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.806 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:42.807 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:42.829 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=1, eventTime=44788759, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.829 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:42.829 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:42.854 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=44788780, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.854 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:42.854 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:42.876 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=2, eventTime=44788812, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.876 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:42.876 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:42.884 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_UP, actionButton=0, id[0]=0, x[0]=821.2396, y[0]=73.74414, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=44788821, downTime=44788222, deviceId=5, source=0x1002 }
06-09 17:13:42.884 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
```

Scrolling the scroll view:
```
06-09 17:13:59.193 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_DOWN, actionButton=0, id[0]=0, x[0]=827.2341, y[0]=62.78412, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=44805126, downTime=44805126, deviceId=5, source=0x1002 }
06-09 17:13:59.193 21164-21164/com.kabouzeid.gramophone.debug D/ACTION_DOWN?: True
06-09 17:13:59.193 21164-21164/com.kabouzeid.gramophone.debug D/Event X: 827
06-09 17:13:59.193 21164-21164/com.kabouzeid.gramophone.debug D/Event Y: 608
06-09 17:13:59.193 21164-21164/com.kabouzeid.gramophone.debug D/View Left: 0
06-09 17:13:59.193 21164-21164/com.kabouzeid.gramophone.debug D/View Right: 0
06-09 17:13:59.194 21164-21164/com.kabouzeid.gramophone.debug D/View Top: 0
06-09 17:13:59.194 21164-21164/com.kabouzeid.gramophone.debug D/View Bottom: 0
06-09 17:13:59.194 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:59.228 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=827.2341, y[0]=62.78412, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=44805154, downTime=44805126, deviceId=5, source=0x1002 }
06-09 17:13:59.228 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:59.229 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:59.244 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=827.2341, y[0]=62.78412, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=1, eventTime=44805175, downTime=44805126, deviceId=5, source=0x1002 }
06-09 17:13:59.244 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:59.244 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:59.260 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=827.2341, y[0]=62.78412, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=1, eventTime=44805196, downTime=44805126, deviceId=5, source=0x1002 }
06-09 17:13:59.260 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:59.260 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:59.276 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=827.2341, y[0]=62.78412, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=2, eventTime=44805212, downTime=44805126, deviceId=5, source=0x1002 }
06-09 17:13:59.276 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:59.276 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 17:13:59.292 21164-21164/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=803.2302, y[0]=57.744385, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=1, eventTime=44805228, downTime=44805126, deviceId=5, source=0x1002 }
06-09 17:13:59.292 21164-21164/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 17:13:59.292 21164-21164/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
```

theduffmaster commented:
I'm guessing the solution has something to do with [this](https://stackoverflow.com/questions/9965695/how-to-distinguish-between-move-and-click-in-ontouchevent). Looks like whether ACTION_MOVE is triggered is device specific.

arkon commented:
@theduffmaster Actually, this is sort of weird. So those logs I posted above are from my OnePlus One (running LineageOS 14.1) and is what I used to record the video too. I just tried it on my Nexus 7 (2013 model, also running LineageOS 14.1), but it works as intended.

theduffmaster commented:
@arkon yep I think it's definitely a device specific thing. I'll see what I can do to sort it out.

theduffmaster commented:
@arkon I believe I fixed it, but of course I can't be sure since it's always worked on my phone. Can you test it?

arkon commented:
@theduffmaster Did you forget to push something?

arkon commented:
@theduffmaster It does log `ACTION_UP click?: true`, but it still doesn't work as intended.

theduffmaster commented:
So the current problem is that now that I intercept the ACTION_UP, it never gets passed to the proper view and Android thinks that the user has just long clicked.

theduffmaster commented:
@arkon try it now. If it works, let me know how the click looks.

arkon commented:
@theduffmaster Still no luck. Here's the logcat from a regular tap on the scroll view:

```
06-09 19:20:28.622 22843-22843/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_DOWN, actionButton=0, id[0]=0, x[0]=753.3025, y[0]=61.49353, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=47900769, downTime=47900769, deviceId=5, source=0x1002 }
06-09 19:20:28.622 22843-22843/com.kabouzeid.gramophone.debug D/ACTION_DOWN?: True
06-09 19:20:28.622 22843-22843/com.kabouzeid.gramophone.debug D/Event X: 753
06-09 19:20:28.622 22843-22843/com.kabouzeid.gramophone.debug D/Event Y: 687
06-09 19:20:28.622 22843-22843/com.kabouzeid.gramophone.debug D/View Left: 172
06-09 19:20:28.622 22843-22843/com.kabouzeid.gramophone.debug D/View Right: 927
06-09 19:20:28.622 22843-22843/com.kabouzeid.gramophone.debug D/View Top: 663
06-09 19:20:28.622 22843-22843/com.kabouzeid.gramophone.debug D/View Bottom: 715
06-09 19:20:28.622 22843-22843/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 19:20:28.653 22843-22843/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=753.3025, y[0]=61.49353, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=47900797, downTime=47900769, deviceId=5, source=0x1002 }
06-09 19:20:28.653 22843-22843/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 19:20:28.653 22843-22843/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 19:20:28.668 22843-22843/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=753.3025, y[0]=61.49353, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=1, eventTime=47900818, downTime=47900769, deviceId=5, source=0x1002 }
06-09 19:20:28.668 22843-22843/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 19:20:28.668 22843-22843/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 19:20:28.687 22843-22843/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=753.3025, y[0]=61.49353, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=47900825, downTime=47900769, deviceId=5, source=0x1002 }
06-09 19:20:28.687 22843-22843/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 19:20:28.687 22843-22843/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 19:20:28.688 22843-22843/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_UP, actionButton=0, id[0]=0, x[0]=753.3025, y[0]=61.49353, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=47900838, downTime=47900769, deviceId=5, source=0x1002 }
06-09 19:20:28.688 22843-22843/com.kabouzeid.gramophone.debug D/ACTION_UP click?: true
06-09 19:20:28.690 22843-22843/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
```

theduffmaster commented:
@arkon see if this works.

arkon commented:
@theduffmaster Regular taps are now registered, although the ripple animation never happens. Long pressing doesn't work.

On my Nexus 7 (where it worked before anyway) the ripple does appear when tapping, but it's slightly delayed.

theduffmaster commented:
@arkon try this. No need to try it on the Nexus 7.

arkon commented:
@theduffmaster Nope, same as before.

theduffmaster commented:
@arkon let me know if this works. If it does work let me know if it longpresses when you scroll.

theduffmaster commented:
When you scroll the textview that is.

arkon commented:
@theduffmaster Yup, it works! And (as you probably expected), it does register as a long press when scrolling for long enough. Perhaps you could calculate the movement delta to determine if it's a scroll or not?

theduffmaster commented:
@arkon Was already on it!

theduffmaster commented:
@arkon That should fix it, let me know if it works. If it works I'm going to try to optimize it a bit so I might need you to test one commit.

arkon commented:
@theduffmaster Unfortunately, that last commit broke it and it's back to not working at all. :(

theduffmaster commented:
@arkon Let me know what your logcat outputs when you click and nothing happens.

arkon commented:
@theduffmaster Apparently it thinks it's scrolling?

```
06-09 20:57:11.165 16370-16370/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_DOWN, actionButton=0, id[0]=0, x[0]=747.30804, y[0]=63.03943, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=704472, downTime=704472, deviceId=5, source=0x1002 }
06-09 20:57:11.166 16370-16370/com.kabouzeid.gramophone.debug D/ACTION_DOWN?: True
06-09 20:57:11.166 16370-16370/com.kabouzeid.gramophone.debug D/Event X: 747
06-09 20:57:11.166 16370-16370/com.kabouzeid.gramophone.debug D/Event Y: 812
06-09 20:57:11.166 16370-16370/com.kabouzeid.gramophone.debug D/View Left: 172
06-09 20:57:11.166 16370-16370/com.kabouzeid.gramophone.debug D/View Right: 927
06-09 20:57:11.166 16370-16370/com.kabouzeid.gramophone.debug D/View Top: 786
06-09 20:57:11.166 16370-16370/com.kabouzeid.gramophone.debug D/View Bottom: 838
06-09 20:57:11.166 16370-16370/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
06-09 20:57:11.196 16370-16370/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=747.30804, y[0]=63.03943, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=704499, downTime=704472, deviceId=5, source=0x1002 }
06-09 20:57:11.196 16370-16370/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 20:57:11.196 16370-16370/com.kabouzeid.gramophone.debug D/Outside Scrollview: True
06-09 20:57:11.196 16370-16370/com.kabouzeid.gramophone.debug D/scrolling: True
06-09 20:57:11.221 16370-16370/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_MOVE, actionButton=0, id[0]=0, x[0]=747.30804, y[0]=63.03943, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=704527, downTime=704472, deviceId=5, source=0x1002 }
06-09 20:57:11.221 16370-16370/com.kabouzeid.gramophone.debug D/ACTION MOVE: True
06-09 20:57:11.221 16370-16370/com.kabouzeid.gramophone.debug D/Outside Scrollview: True
06-09 20:57:11.221 16370-16370/com.kabouzeid.gramophone.debug D/scrolling: True
06-09 20:57:11.232 16370-16370/com.kabouzeid.gramophone.debug D/Touch Event Intercepted: MotionEvent { action=ACTION_UP, actionButton=0, id[0]=0, x[0]=747.30804, y[0]=63.03943, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x2, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=704541, downTime=704472, deviceId=5, source=0x1002 }
06-09 20:57:11.232 16370-16370/com.kabouzeid.gramophone.debug D/InterceptTouch Finished: True
```

theduffmaster commented:
That is so weird

theduffmaster commented:
@arkon Try playing around with [this](https://github.com/kabouzeid/Phonograph/pull/160/commits/f5e4aba8af6b578b42cf709537753d5a166eb40d#diff-852fe1e157281bc311774c57992b79dbR27) value and maybe something works. Try some really high and really low values.

arkon commented:
@theduffmaster Is there somewhere that I can just chat with you (like Hangouts) so we don't blow up this PR thread/people's inboxes?

theduffmaster commented:
Good idea. We could do hangouts if that works for you. What's the best way to add you?

arkon commented:
@theduffmaster You can find my email on my website (which is on my GitHub profile).

theduffmaster commented:
@arkon added you

arkon commented:
@theduffmaster I'm not seeing anything. Did you start a chat? You could also add me on Facebook (same name/profile picture) if that's easier.

theduffmaster commented:
It should work on all devices now. Big thank you to @arkon for figuring out how to get it working for devices that call ACTION_MOVE during a tap action.

arkon commented:
I think the preferable way to handle this might just be the same thing as Google Play Music: truncate text and show ellipses as before, but having automatically scrolling text for the currently playing track (e.g. in the collapsed bar at the bottom, and at the top of the queue).

theduffmaster commented:
I believe I've fixed everything. Please test it.
For me at least this is a major feature. The space provided to show names is quite small and only gets smaller on "small" screen phones like my M8. The majority of my albums and songs have names longer then the space provided. If the name is something like "…50th Anniversary Edition", "…Remix", "…Disk 2", "…Take 15" or "…Feat. …" I need to be able to see what the whole name is. Switching my phone to landscape doesn't fix anything because it just divides the playing queue and the playing controls to each cover half a screen. Opening the tag editor is really cumbersome. This lets users check the full title of any song they want, and it doesn't use the song scrolling view which I think looks terrible, like something out of some website made in 1999, and also doesn't allow the user full control.

kabouzeid commented:
Alright I'll have a look at this in the evening, thanks for your effort!

arkon commented on app/src/main/java/com/kabouzeid/gramophone/views/TouchInterceptFrameLayout.java:
```diff
@@ -0,0 +1,199 @@
+     * @param sv The HorizontalScrollView containing text that needs to be scrolled
+     * @param tv The TextView that needs to be scrolled (typically song or album title)
+     */
+    public void setScrollViews(HorizontalScrollView sv, TextView tv) {
```
Might be better to just call this `setViews`.

theduffmaster replied:
Good call

arkon replied:
I'm also thinking that there might be a better name for this than `TouchInterceptFrameLayout` since it pretty specifically handles the truncated/scrolling text, but I'm not sure what exactly to call it. I'm also sort of wondering if this could actually be simplified even further such that the `TextView` doesn't even need to be manually added in the XML.

theduffmaster replied:
I kept the textview so you could just surround any textview in a view group with a horizontal scrollview, but maybe there's a better way… The name does definitely need to change though, maybe something like FrameLayoutWithScrollingChild. Not that, but something in the same sense.

arkon commented on app/src/main/java/com/kabouzeid/gramophone/views/TouchInterceptFrameLayout.java:
```diff
@@ -0,0 +1,199 @@
+    /**
+     * Fetches the HorizontalScrollView contained by this FrameLayout and also gets the
+     * TextView that is contained within that HorizontalScrollView.
+     * Must be called before setTruncateText.
```
Why not just merge the two then? You could pass in the text to this method as well.

theduffmaster replied:
For some classes you might need to conditionally set the truncate text separately, perhaps multiple times, so I made it a separate class.

arkon commented on app/src/main/res/values/strings.xml:
```diff
@@ -298,4 +298,6 @@
     <string name="app_shortcut_last_added_long">@string/last_added</string>
     <string name="app_shortcut_last_added_short">@string/last_added</string>
     <string name="playlist_is_empty">Playlist is empty</string>
+    <string name="NULL_VIEWS_EXCEPTION_MESSAGE">Did you forget to call setScrollViews when creating your FrameLayout? Either textView or scrollView is null.</string>
```
These don't need to be added to the XML files, they can just be in the Java files.

theduffmaster replied:
I guess since we're not using them again no need to store them.

arkon commented on app/src/main/java/com/kabouzeid/gramophone/views/TouchInterceptFrameLayout.java:
```diff
@@ -0,0 +1,199 @@
+                                Log.e("E/TouchInterceptFL","songTruncated is empty or null. Did you remember " +
+                                        "to set the song string when setting the song name in your text view?");
+                            }
+                            cancelPendingInputEvents();
```
This chunk of code is identical to the block of code for handling vertical scrolling below. Might be better to just pull it out into a private method or something.

theduffmaster replied:
Done

arkon commented on app/src/main/java/com/kabouzeid/gramophone/views/TouchInterceptFrameLayout.java:
```diff
@@ -0,0 +1,199 @@
+                        }
+                    }
+                    // Scrolling vertically: cancel horizontal scrolling events
+                    if (distanceY > MAX_VERTICAL_DISTANCE) {
```
Might as well do this check before and if so, prematurely break out of this?

kabouzeid commented:
It doesn't work for me: Initially the text it is truncated with "...", when I scroll the text around and then back to the initial state the "..." doesn't appear again. Also sometimes the songs stays in a somehow "selected" state where there ripple doesn't disappear.

Google Pixel, Android 7.1.2

kabouzeid commented:
Also I don't think it's good for UX when the user wants to switch tabs by swiping and instead the song title scrolls.

kabouzeid commented:
I still think it would be best to just put the full song title in the detail dialog.

arkon commented:
I still think we should just automatically scroll the song title if it's too long in the now playing screen like Google Play Music. We can also make it so that the toolbar has a scrollable title when a track is long pressed in the lists, so a user can long press an item then scroll the full title there (using the same mechanism that this PR currently has).

theduffmaster commented:
The last commit had some problems with how I handled long clicks so that's
why it stays selected. I'll fix that once I get home from work. If anything
I think we should at least make this an experimental feature that can be
user activated. I know I definitely would use this all the time.
Personally, I'm against any automatically scrolling text. In my opinion it
looks very tacky and out of place compared to the rest of the ui. It also
doesn't do much for the user. I usually know what song I'm playing. What I
do want to know is what song am I looking at right now and possibly playing.

On Mon, Jun 12, 2017, 4:37 PM Eugene Cheung <notifications@github.com>
wrote:

> I still think we should just automatically scroll the song title if it's
> too long in the now playing screen like Google Play Music. We can also make
> it so that the toolbar has a scrollable title when a track is long pressed
> in the lists, so a user can long press an item then scroll the full title
> there (using the same mechanism that this PR currently has).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kabouzeid/Phonograph/pull/160#issuecomment-307916756>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGCrndLyf3qEjPJCoUvpfvJwGmo8j9xBks5sDaGfgaJpZM4NsM2W>
> .
>

kabouzeid commented:
I'm also against autoscrolling. An option would add even more complications to the feature. The toolbar scrolling is a good idea, however the space in the toolbar is even more limited and no one would expect it to be there.

arkon commented:
FYI @theduffmaster there's merge conflicts for this branch right now.

theduffmaster commented:
@arkon It's just a conflict in the imports. Those happen sometimes, it shouldn't be a problem.

arkon commented:
@theduffmaster Regardless, it'll need to be resolved otherwise it can't be merged.

I'll test this out later when I have time though. 👍

theduffmaster commented:
@arkon @kabouzeid I'm pretty confident that you'll find it much more usable and unobtrusive this time around. When you find some time, please test it out and let me know how it goes.

arkon commented:
@theduffmaster I do like how it reverts back to its original look after some time. However, I'm seeing some weird behaviour sometimes. You can see it [in this video](https://www.youtube.com/watch?v=eG6uwqjhPRA) (might need to wait for it to be processed), where long-pressing an item where the `TextView` is doesn't quite work sometimes (it's like it works, then gets immediately cancelled).

It's also a bit weird that you can scroll on the now playing bar at the bottom, because swiping horizontally should change the track.

theduffmaster commented:
@arkon I didn't realize the mini player swiped, I'll revert that. You're right, the long click is getting canceled. I forgot about the MotionEvent.Move getting triggered even on a tap. That should be easy to fix, once I get home I'll push a fix for that. Anything else that's weird on your side?

arkon commented:
@theduffmaster I haven't noticed anything else yet from my minimal testing so far. I'll continue using it tomorrow during my commute and see if anything crops up. :)

theduffmaster commented:
@arkon When you get a chance let me know if this fixes it.

arkon commented:
@theduffmaster I'm still seeing it. It mostly happens on list items where the track title _isn't_ long enough to be truncated, where you long press in an empty space where longer text _would_ go.

I also noticed that if you immediately trying scrolling the text again right after it jumps back, it'll sort of fight you and jump back anyway. I haven't really looked at the code yet, but we need to make sure that a "currently running" jump back is cancelled if the user initiates another scroll.

theduffmaster commented:
@arkon Huh, that really sucks. I actually spent a good amount of time making sure it doesn't jump back if the user tries to scroll again. It's set so if the user touches it again it will cancel the scroll back. 

So for the long clicking is long clicking working on all truncated text but not working on only non-truncated text? If that's the case then I can fix that easily.

theduffmaster commented:
@arkon Long click should be fixed now. Turns out I wasn't dealing with a null String correctly.

arkon commented:
@theduffmaster Long clicks seem fine now. The only issue with it is that the highlight state sometimes remains after long clicking (but not always).

Another thing is that selecting an item (i.e. long clicking it) then unselecting it causes the RecyclerView to refresh, so there's brief flashes on long text items where you can see it's not truncated then it gets the truncated text set to them. This might be fixed by keeping them truncated via XML, then programmatically setting it so it's not on init?

arkon commented:
@theduffmaster On the note of the scroll back fighting: it was a pretty forced edge case (very intentionally trying to get it to fight me), so I wouldn't be that concerned with it.

theduffmaster commented:
@arkon I totally forgot that I actually spent some time trying to get the set truncating via XML working, but unfortunately text won't automatically truncate if it's in a ScrollView which is why I had to figure out how to do it programmatically. It shouldn't be too hard though to figure out how to fix the flashing.

theduffmaster commented:
@arkon @kabouzeid So a couple of changes, for the better of course. Now there's no need to set anything anywhere. Just add the custom views as you usually would any custom view and it all works from the go. That means it can be used on any textview very easily. Also, now the truncate text is the initial text set, so no flashes or anything. It's as if you set `ellipsize:true` via xml. Furthermore, the implementation is a lot simpler now and I believe it works a lot better. Lastly, it will now always retruncate and I fixed the occasionally having to fight with it after touching it again and trying to scroll it. When you get a chance, please test it and let me know how it goes.

arkon commented on .gitignore:
```diff
@@ -40,3 +40,5 @@ captures/
 # Mac
 .DS_Store
+
+app/build.gradle
```
This shouldn't be committed.

theduffmaster replied:
Sorry, for some reason my gitignore keeps getting messed up.

arkon commented on app/build.gradle:
```diff
@@ -119,4 +119,4 @@ dependencies {
     compile 'com.jakewharton:butterknife:8.6.0'
     annotationProcessor 'com.jakewharton:butterknife-compiler:8.6.0'
-}
+}
```
Keep the newline here.

theduffmaster replied:
Sorry, for some reason my gitignore keeps getting messed up.

arkon commented on app/src/main/java/com/kabouzeid/gramophone/adapter/SearchAdapter.java:
```diff
@@ -43,6 +43,8 @@
     private final AppCompatActivity activity;
     private List<Object> dataSet;
 
+    private RecyclerView recyclerView;
```
Is this used...?

theduffmaster replied:
Nope, removed it.

arkon commented on app/src/main/java/com/kabouzeid/gramophone/adapter/base/MediaEntryViewHolder.java:
```diff
@@ -57,6 +68,10 @@ public MediaEntryViewHolder(View itemView) {
         super(itemView);
         ButterKnife.bind(this, itemView);
 
+        if (title_scrollview != null && title != null && itemView instanceof TouchInterceptFrameLayout) {
```
Remove this if it's not needed.

theduffmaster replied:
Sorry again, thought I had removed that already.

arkon commented on app/src/main/res/layout/item_grid.xml:
```diff
@@ -36,13 +37,19 @@
 
-            <TextView
-                android:id="@+id/title"
+            <com.kabouzeid.gramophone.views.TouchInterceptHorizontalScrollView
```
I'm not a fan of having this in the grid. It's a bit janky when you sometimes try to swipe to a different tab but end up scrolling a name in a small grid item instead. The user can always open it up and scroll it there anyway. :)

theduffmaster replied:
Agreed, reverted it.

arkon commented on app/src/main/res/layout/item_list.xml:
```diff
@@ -20,7 +22,15 @@
         app:srcCompat="@drawable/ic_drag_vertical_white_24dp"
         tools:ignore="ContentDescription" />
 
+    <View
```
You should put a comment here for future reference so people know what this is for.

theduffmaster replied:
I plan on going through this and documenting everything tonight, but I'll definitely do this.

theduffmaster replied:
@arkon Let me know if there's anything you still see that might not be so clear. I'm pretty sure I explained everything, but I'd be surprised if I didn't miss a thing or two.

arkon replied:
I'll look through the view classes tomorrow evening. 👍

arkon replied:
So what is this invisible view for again?

arkon commented on app/src/main/res/values/attrs.xml:
```diff
@@ -12,4 +12,13 @@
     <attr name="dividerColor" format="color" />
 
+    <!-- Setting Views. -->
+    <declare-styleable name="TouchInterceptFrameLayout">
```
This isn't necessary anymore, right?

theduffmaster replied:
Nope, removed.

arkon commented on app/src/main/java/com/kabouzeid/gramophone/views/TouchInterceptFrameLayout.java:
```diff
@@ -0,0 +1,162 @@
+
+    public TouchInterceptFrameLayout(@NonNull Context context) {
+        this(context, null);
+        setTag("TIFL");
```
These should be extracted as `static final` variables.

arkon commented on app/src/main/java/com/kabouzeid/gramophone/views/TouchInterceptTextView.java:
```diff
@@ -0,0 +1,193 @@
+        String currentText = getText().toString();
+        Boolean isUntruncatedSong = currentText.endsWith("\uFEFF");
+
+        if(!currentText.endsWith("\u202F") &&
```
You should put a comment saying that `"\u202F"` is (I assume) ellipses.

theduffmaster replied:
It's actually not an ellipses, but I should put something there to make it more clear what it is. It's an invisible unicode character I'm using to make what's been truncated.

arkon commented on app/src/main/java/com/kabouzeid/gramophone/views/TouchInterceptHorizontalScrollView.java:
```diff
@@ -0,0 +1,279 @@
+     *
+     * @param enable Set true to enable dragging, false to disable
+     */
+    public void slidingPanelSetTouchEnabled(boolean enable) {
```
I'm not a fan of having this in here. Ideally, these view classes should be totally independent of whatever context they're in.

theduffmaster replied:
It doesn't seem like it even does it's job very well so I'm going to remove it.

arkon commented on app/src/main/java/com/kabouzeid/gramophone/views/TouchInterceptHorizontalScrollView.java:
```diff
@@ -0,0 +1,279 @@
+     */
+    public void CancelClick() {
+        getRootView().cancelLongPress();
+        getRootView().cancelPendingInputEvents();
```
`cancelPendingInputEvents()` requires API 19 but the current minimum is 16. You should wrap it in an API version check, but also look into how it affects lower APIs without it.

theduffmaster replied:
I've decided to take this out as well, since it was originally added as a fail safe but it seems the initial checks for whether its a scroll or a tap work so well that we don't need this anymore.

arkon commented on app/src/main/res/layout/activity_artist_detail.xml:
```diff
@@ -32,25 +34,38 @@
         android:orientation="vertical">
 
-        <TextView
+        <View
```
What's this for again?

theduffmaster replied:
These were for testing and I forgot to remove them. Removed.

arkon commented on app/src/main/java/com/kabouzeid/gramophone/views/TouchInterceptFrameLayout.java:
```diff
@@ -0,0 +1,132 @@
+            switch (e.getAction() & MotionEvent.ACTION_MASK) {
+                case MotionEvent.ACTION_DOWN:
+                    if (!touchedScrollView) {
+                        scrollView.cancelPendingInputEvents();
```
Is this required? It needs minimum API 19.

theduffmaster replied:
I guess I missed this one when looking it over. Unfortunately this one is required. I've implemented a API check so that if the API is under 19 it will invoke scrollView.cancelLongPress() instead. That means that below API 19 if you really mess with it you can get the background of the view to turn grey.

theduffmaster commented:
@arkon I removed the invisible view (they were originally there for testing) and changed the text view name to AutoTruncateTextView, I agree a much more suitable name. It seems like at some point the text setting was broken and it now occasionally sets the text in the recyclerview as the previous text. I'm on vacation, but when I get a chance I'll look into this. It wasn't happening before so I wonder what broke it.

theduffmaster commented:
It seems the invisible views served to keep the artist page visible. I'll
add them back when I get to a computer.

On Sat, Jul 29, 2017, 7:05 PM Eugene Cheung <notifications@github.com>
wrote:

> *@arkon* commented on this pull request.
>
> Looks pretty good to me. I'd consider maybe renaming the TextView to
> something like AutoTruncateTextView though, since that's more clear as to
> what it does. Moving the three view classes to a package can keep it
> obvious that they work with each other too.
> ------------------------------
>
> In app/src/main/res/layout/item_list.xml
> <https://github.com/kabouzeid/Phonograph/pull/160#discussion_r130230047>:
>
> > @@ -20,7 +22,15 @@
>          app:srcCompat="@drawable/ic_drag_vertical_white_24dp"
>          tools:ignore="ContentDescription" />
>
> +    <View
>
> So what is this invisible view for again?
> ------------------------------
>
> In app/src/main/res/layout/activity_artist_detail.xml
> <https://github.com/kabouzeid/Phonograph/pull/160#discussion_r130230053>:
>
> > @@ -32,25 +34,38 @@
>          android:layout_height="wrap_content"
>          android:orientation="vertical">
>
> -        <TextView
> +        <View
>
> What's this for again?
> ------------------------------
>
> In
> app/src/main/java/com/kabouzeid/gramophone/views/TouchInterceptFrameLayout.java
> <https://github.com/kabouzeid/Phonograph/pull/160#discussion_r130230063>:
>
> > +    public boolean onInterceptTouchEvent(MotionEvent e) {
> +        int x = Math.round(e.getRawX());
> +        int y = Math.round(e.getRawY());
> +
> +        scrollView = getTouchInterceptHorizontalScrollView();
> +        scrollView.getGlobalVisibleRect(scrollViewRect);
> +
> +        boolean touchedScrollView =
> +                x > scrollViewRect.left && x < scrollViewRect.right &&
> +                y > scrollViewRect.top && y < scrollViewRect.bottom;
> +
> +        if (scrollView.isScrollable()) {
> +            switch (e.getAction() & MotionEvent.ACTION_MASK) {
> +                case MotionEvent.ACTION_DOWN:
> +                    if (!touchedScrollView) {
> +                        scrollView.cancelPendingInputEvents();
>
> Is this required? It needs minimum API 19.
>
> —
> You are receiving this because you were mentioned.
>
> Reply to this email directly, view it on GitHub
> <https://github.com/kabouzeid/Phonograph/pull/160#pullrequestreview-53085451>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGCrnTD3To0-IRKgtVYH6ftcFNCLN761ks5sS7qvgaJpZM4NsM2W>
> .
>

arkon commented:
I'm noticing the occasional wrong string too.

arkon commented:
@theduffmaster I committed something that I _think_ is a better fix for it. Let me know if you still see the issue.

theduffmaster commented on app/src/main/java/com/kabouzeid/gramophone/views/AutoTruncateTextView.java:
```diff
@@ -115,24 +115,24 @@ protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
+        }
+        this.truncatedText = truncatedText;
+
+        final TouchInterceptHorizontalScrollView scrollView = getTouchInterceptHorizontalScrollView();
```
As long as the scrollview never shows up as null, which I don't think it will, this will do just fine. Good work!

theduffmaster commented:
@arkon After a couple days of testing your fix seems to work! Thanks again. Pending a couple more days of testing I think this is about done.
